test/test_examples.py

```diff
@@ -2,118 +2,28 @@
 (Could be seen as a kind of integration test.)
 
 """
-import contextlib
 import glob
-import importlib
-import shlex
 import sys
-from io import StringIO
-from typing import Callable, Optional
 from pathlib import Path
-import pytest
 
-expected = ""
-
-
-def xfail_param(*args, reason: str):
-    return pytest.param(*args, marks=pytest.mark.xfail(reason=reason))
-
-
-@pytest.fixture
-def set_prog_name():
-    argv = sys.argv.copy()
-    del sys.argv[1:]
-
-    def set_prog(prog_name: str, args: str):
-        sys.argv[0] = prog_name
-        sys.argv[1:] = shlex.split(args)
-
-    yield set_prog
-    sys.argv = argv
-
-
-import textwrap
-
-
-@pytest.fixture
-def assert_equals_stdout(capsys):
-    def strip(string):
-        return "".join(string.split())
-
-    import difflib
+tests_to_run = (
+    p for p in glob.glob("examples/**/*.py") if "custom_types_example" not in p
+)
 
-    def should_equal(expected: str, file_path: str):
-        out = capsys.readouterr().out
-        # assert strip(out) == strip(expected), file_path
-        # assert out == expected, file_path
-        out_lines = out.splitlines(keepends=False)
-        out_lines = [line.strip() for line in out_lines if line and not line.isspace()]
-        expected_lines = expected.splitlines(keepends=False)
-        expected_lines = [
-            line.strip() for line in expected_lines if line and not line.isspace()
-        ]
-        assert out_lines == expected_lines, file_path
 
-    return should_equal
+def pytest_generate_tests(metafunc):
```

Comment:
As mentioned above, while this is clever from a testing point of view, I don't want assertions in the examples. Please revert all of this.